### PR TITLE
feat(confidential/builder): add GetSpendingBalance ledger reader

### PR DESCRIPTION
**Pull request by aluque-peersyst**
# feat(confidential/builder): add GetSpendingBalance ledger reader

## Description
This PR aims to add `GetSpendingBalance` to `confidential/builder`, a read-only helper that resolves a holder's `ConfidentialBalanceSpending` from one validated ledger and decrypts it with the holder's ElGamal private key. It takes the same `LedgerQuerier` as the builders, so `rpc.Client` and `websocket.Client` share one reader instead of every caller combining `GetLedgerEntry` and `elgamal.Decrypt` by hand.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Documentation update
- [x] Refactoring

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes

## Changes

- Add `GetSpendingBalance` and `SpendingBalanceParams`. Both ledger reads go through the existing validated-ledger snapshot; no account sequence is queried.
- Return `ErrMPTokenNotFound` when the holder has no `MPToken`, and zero without decrypting when the `MPToken` has no spending ciphertext.
- Exclude `ConfidentialBalanceInbox`, which is unspendable until a `ConfidentialMPTMergeInbox`.
- Cap `BalanceRange` at the issuance `ConfidentialOutstandingAmount` via a new `boundedBalanceRange`, now shared with `BuildClawback`.
- Add unit tests (CGo and non-CGo), document the helper in the builders guide, update the RPC and WebSocket confidential examples, and add a changelog entry.

### CODE REVIEW

File: examples/confidential/ws/main.go [hunk 1]
```diff
@@ -297,7 +297,12 @@ func report(client *websocket.Client, address types.Address, key elgamal.Keypair
 	if err != nil {
 		return fmt.Errorf("%s inbox: %w", label, err)
 	}
-	spending, err := decrypt(token.ConfidentialBalanceSpending, key.PrivKeyHex)
+	spending, err := builder.GetSpendingBalance(client, builder.SpendingBalanceParams{
```
Comment (kpitapeersyst) [inline examples/confidential/ws/main.go]: We already fetch the MPToken here, so calling `GetSpendingBalance` fetches it again. If the state changes between reads, we could print inconsistent values. Keeping the old pattern saves a request and keeps everything we print from the same ledger.

File: examples/confidential/rpc/main.go [hunk 2]
```diff
@@ -291,7 +291,12 @@ func report(client *rpc.Client, address types.Address, key elgamal.Keypair, labe
 	if err != nil {
 		return fmt.Errorf("%s inbox: %w", label, err)
 	}
-	spending, err := decrypt(token.ConfidentialBalanceSpending, key.PrivKeyHex)
+	spending, err := builder.GetSpendingBalance(client, builder.SpendingBalanceParams{
```
Comment (kpitapeersyst) [inline examples/confidential/rpc/main.go]: We already fetch the MPToken here, so calling `GetSpendingBalance` fetches it again. If the state changes between reads, we could print inconsistent values. Keeping the old pattern saves a request and keeps everything we print from the same ledger.

File: confidential/builder/balance.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	// ConfidentialOutstandingAmount.
+	BalanceRange elgamal.AmountRange
+}
+
+// GetSpendingBalance reads a holder's ConfidentialBalanceSpending from one validated ledger
+// and decrypts it with the holder's ElGamal private key. The inbox is not included, because
+// it cannot be spent until a ConfidentialMPTMergeInbox moves it. An MPToken with no spending
+// ciphertext reads as zero without decrypting. Decryption requires CGo.
+//
+// Errors: ErrMPTokenNotFound, ErrIssuanceNotFound, ErrLedgerQuery, ErrInvalidLedgerState,
```
Comment (kpitapeersyst) [inline confidential/builder/balance.go]: Could we expand this error list or clarify that it isn’t exhaustive? The function can also return validation errors, `ErrConfidentialDisabled`, and `elgamal.ErrInvalidAmountRange`.

File: confidential/builder/balance_test.go [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+	return SpendingBalanceParams{
+		Holder:        testAccount,
+		IssuanceID:    testIssuanceID,
+		HolderPrivKey: testHolderPrivKey,
+		BalanceRange:  elgamal.AmountRange{Low: 0, High: 1000},
+	}
+}
+
+// unspendableCiphertext is a well-formed ciphertext that fails to decrypt.
+const unspendableCiphertext = "not-a-ciphertext"
```
Comment (kpitapeersyst) [inline confidential/builder/balance_test.go:49]: Small nit: `"not-a-ciphertext"` is malformed, so the comment should say that rather than `“well-formed.”`

File: confidential/builder/balance_crypto_test.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	params.HolderPrivKey = holderKP.PrivKeyHex
+	params.BalanceRange = elgamal.AmountRange{Low: 0, High: spendingBalance}
+
+	balance, err := GetSpendingBalance(q, params)
+	require.NoError(t, err)
+	require.Equal(t, spendingBalance, balance)
+	require.Empty(t, q.accountRequests, "reading a balance needs no account sequence")
+}
+
+func TestGetSpendingBalanceDecryptsEncryptedZero(t *testing.T) {
```
Comment (kpitapeersyst) [inline confidential/builder/balance_crypto_test.go:42]: Could we also test encrypted zero with `ConfidentialOutstandingAmount` omitted and a nonzero upper search bound? That would check the full path where the omitted supply becomes zero, caps the range to `[0,0]`, and decryption still succeeds.

File: confidential/builder/balance_crypto_test.go [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+	params := spendingBalanceParams()
+	params.HolderPrivKey = holderKP.PrivKeyHex
+	params.BalanceRange = elgamal.AmountRange{Low: 0, High: 0}
+
+	balance, err := GetSpendingBalance(q, params)
+	require.NoError(t, err)
+	require.Zero(t, balance)
+}
+
+func TestGetSpendingBalanceSearchesWiderRangesUpToTheSupply(t *testing.T) {
```
Comment (kpitapeersyst) [inline confidential/builder/balance_crypto_test.go]: The wide-range test would pass even without capping. It tests successful decryption, not the cap itself. Separate unit tests already verify the cap.

**Review by kpitapeersyst — CHANGES_REQUESTED** [review]
Good job!

Just a few minor changes requested.

**Review by kpitapeersyst — APPROVED** [review]
LGTM! 🚀
